data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -210,6 +221,72 @@ impl TraceExporter {
         TraceExporterBuilder::default()
     }
 
+    fn runtime(&self) -> Result<Arc<Runtime>, TraceExporterError> {
+        match self.runtime.lock_or_panic().as_ref() {
```

Comment:
> [!WARNING]
> Could we not cause panic here? 

You can safely unwrap and return a result here, similarly at other places

Reply:
It's a mutex lock. The only time it's going to panic is if another thread holding the lock panics. Practically speaking, if we are using `panic_unwind`, I'm not sure what benefit we get out of handling the error here.